osu.Game/Beatmaps/BeatmapManager.cs

```diff
@@ -307,6 +307,17 @@ private ArchiveReader getReaderFrom(string path)
         /// <returns>The imported beatmap, or an existing instance if it is already present.</returns>
         private BeatmapSetInfo importToStorage(ArchiveReader reader)
         {
+            string mapName;
+            // let's make sure there are actually .osu files to import.
+            try
+            {
+                mapName = reader.Filenames.First(f => f.EndsWith(".osu"));
+            }
+            catch
+            {
+                throw new InvalidOperationException("No beatmap files found in the map folder.");

```

Comment:
These are all caught further up the stack in `Import()`. Maybe this isn't ideal, but I'm following existing conventions for this section of the code (see `BeatmapDecoder` and `ConvertHitObjectParser`)